src/Razor/src/Microsoft.VisualStudio.LegacyEditor.Razor/Parsing/VisualStudioRazorParser.cs

```diff
@@ -561,10 +562,10 @@ private void CompleteCodeDocumentRequestsForSnapshot(RazorCodeDocument codeDocum
             }
 
             // The matching requests are in reverse order so we need to invoke them from the back to front.
-            for (var i = matchingRequests.Count - 1; i >= 0; i--)
+            foreach (var request in matchingRequests)
```

Comment:
This now loops forward, not backward, and I doubt our legacy editor testing is up to the task of finding issues here. Perhaps use a ListPool and revert the loop changes?